packages/myst-toc/tests/examples.spec.ts

```diff
@@ -1,49 +1,121 @@
-import { describe, test, expect } from 'vitest';
-import fs from 'node:fs';
-import path from 'node:path';
-import yaml from 'js-yaml';
```

Comment:
Would love to see these remain as yaml tests if possible, much easier to read/change/update, not a big deal though.

Reply:
I usually feel that way. However, here I like the ease of parametrisation and the complexity (match error vs match result etc) is such that I think code is more maintainable.